test/e2e/prometheus-metrics-test

```diff
@@ -145,47 +145,25 @@ echo "✅ Port-forwarded pod $POD_NAME"
 
 sleep 10
 
-for i in $(seq 1 10); do 
+for i in `seq 1 $TAINT_CHECK_CYCLES`; do
     METRICS_RESPONSE=$(curl -L localhost:7000/metrics)
     echo "✅ Fetched /metrics."
-
-    if [[ $METRICS_RESPONSE == *"cordon-and-drain"* ]]; then
-        echo "✅ Metric cordon-and-drain!"
-    else
-        echo "❌ Failed checking metric for cordon-and-drain"
-        EXIT_STATUS=3
-    fi
-
-    if [[ $METRICS_RESPONSE == *"pre-drain"* ]]; then
-        echo "✅ Metric pre-drain!"
-    else
-        echo "❌ Failed checking metric for pre-drain"
-        EXIT_STATUS=3
-    fi
-
-    if [[ $METRICS_RESPONSE == *"runtime_go_gc"* ]]; then
-        echo "✅ Metric runtime_go_gc!"
-    else
-        echo "❌ Failed checking runtime_go_gc metric"
-        EXIT_STATUS=3
-    fi
-
-    if [[ $METRICS_RESPONSE == *"runtime_go_goroutines"* ]]; then
-        echo "✅ Metric runtime_go_goroutines!"
-    else
-        echo "❌ Failed checking runtime_go_goroutines metric"
-        EXIT_STATUS=3
-    fi
-
-    if [[ $METRICS_RESPONSE == *"runtime_go_mem"* ]]; then
-        echo "✅ Metric runtime_go_mem!"
-    else
-        echo "❌ Failed checking runtime_go_mem metric"
-        EXIT_STATUS=3
+    failed=""
+    for METRIC in cordon-and-drain pre-drain runtime_go_gc runtime_go_goroutines runtime_go_mem; do
+        if [[ $METRICS_RESPONSE == *"$METRIC"* ]]; then
+            echo "✅ Metric $METRIC!"
+        else
+            echo "⚠️  Metric $METRIC"
+            failed=$METRIC
+            break
+        fi
+    done
+    if [ -z $failed ]; then
+        exit 0
     fi
-
-    sleep 10
-    [[ $EXIT_STATUS -ne 0 ]] || break
```

Comment:
Any failed metric would cause the loop to exit here on the first iteration, so there really wasn't a retry loop. Fixed that.

Reply:
thanks!